### PR TITLE
Optimize the lookup operation for all file systems

### PR DESCRIPTION
**Pull request by lucassong-mh**

### CODE REVIEW

**Comment by lucassong-mh** [comment]
@boterinas ready
@boterinas label O-test

File: kernel/src/fs/path/mount.rs [hunk 1]
```diff
@@ -8,28 +10,28 @@ use crate::{
...
-    mountpoint_dentry: RwLock<Option<Arc<Dentry_>>>,
+    mountpoint_dentry: RwMutex<Option<Arc<Dentry_>>>,
     /// The associated FS.
     fs: Arc<dyn FileSystem>,
     /// The parent mount node.
-    parent: RwLock<Option<Weak<MountNode>>>,
+    parent: RwMutex<Option<Weak<MountNode>>>,
     /// Child mount nodes which are mounted on one dentry of self.
-    children: Mutex<BTreeMap<DentryKey, Arc<Self>>>,
+    children: RwMutex<HashMap<DentryKey, Arc<Self>>>,
```
Comment (tatetian) [inline kernel/src/fs/path/mount.rs:25]: Read-write locks should only be used when we are certain that writes lock are seldomly acquired. But creating files could be quite frequent for some workloads. So in the case of `children`, I guess using `Mutex` is more appropriate than `RwMutex`. Do you have any other considerations?

Comment (lucassong-mh) [reply kernel/src/fs/path/mount.rs:25]: I actually think through this problem. I think this `children` in `MountNode` can be considered read-most. Acquiring its write lock only appears in `mount()`/`umount()` (not file creation/deletion), which is much less rare than file manipulates. While acquiring its read lock appears in every lookup in the `Dentry` of the root `MountNode`.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**